### ISSUE TITLE
Remove `npm install next-on-netlify`

### ISSUE THREAD

**Issue by ehmicky** [netlify/netlify-plugin-nextjs#10]
`next-on-netlify` is already a `dependencies` of this plugin.

https://github.com/netlify/netlify-plugin-nextjs/blob/ef4529016ed0f87ae89d5f312dc8f8f08d1ef7b6/package.json#L28

https://github.com/netlify/netlify-plugin-nextjs/blob/ef4529016ed0f87ae89d5f312dc8f8f08d1ef7b6/index.js#L6

https://github.com/netlify/netlify-plugin-nextjs/blob/ef4529016ed0f87ae89d5f312dc8f8f08d1ef7b6/index.js#L92

Should the following statement be removed?

https://github.com/netlify/netlify-plugin-nextjs/blob/ef4529016ed0f87ae89d5f312dc8f8f08d1ef7b6/index.js#L31

Note: `@latest` is the default npm tag, so is not required here.

**Comment by lindsaylevine** [comment]
@ehmicky this is a bug jason and i discovered monday night. without running the command on line 31, the function bundling errors out because it's unable to find next-on-netlify dependencies.

**Comment by ehmicky** [comment]
If `next-on-netlify` is a dependency of this plugin, calling `require('next-on-netlify')` should work.
Which error message did you get?

**Comment by lindsaylevine** [comment]
`cannot find module next-aws-lambda` traced to one of the generated functions by next-on-netlify. do you mean like a plain `require('next-on-netlify')` at the top of the file separate from `const nextOnNetlify = require('next-on-netlify');`?

**Comment by ehmicky** [comment]
The problem is the following.
`next-on-netlify` is copying Next.js serverless files to the Netlify Functions directory, renaming each of them to `nextJsPage.js` then wrapping them with the [following small file](https://github.com/netlify/next-on-netlify/blob/main/lib/templates/netlifyFunction.js).
That wrapper file requires `next-aws-lambda`:

https://github.com/netlify/next-on-netlify/blob/a48735f625cc7ad1a080b322673d3e9a11f42d27/lib/templates/netlifyFunction.js#L5

When `zip-it-and-ship-it` bundles Netlify Functions, we look for Functions dependencies in the site's `node_modules`. We do not look for them in each plugin's `node_modules` (which is a separate directory). Therefore, in order to bundle the above Functions correctly, `next-aws-lambda` must be available in the site's `dependencies`.

The above workaround works because `next-aws-lambda` is a direct dependency of `next-on-netlify`: 

https://github.com/netlify/next-on-netlify/blob/a48735f625cc7ad1a080b322673d3e9a11f42d27/package.json#L56

However, there is no need for the site to install all of `next-on-netlify` to make this work. Installing only `next-aws-lambda` should work, and would be faster.

That being said, running `npm install next-on-netlify` is not an optimal solution. It makes this plugin much slower. It might make the build fail (`npm install` fails actually fairly often in our build system). It might not work for Yarn users. The real solution here would be for `next-on-netlify` to bundle Functions (as opposed to rely on `zip-it-and-ship-it`) as suggested by @erezrokah in [this comment](https://github.com/netlify/next-on-netlify-enterprise/issues/9#issuecomment-725596336) and make sure `next-aws-lambda` is included during bundling.

I would suggest the two following action points to solve this issue:
  1. First, as a temporary improvement, replace `npm install next-on-netlify@latest` by `npm install next-aws-lambda`
  2. Then, as a more robust solution, bundle Functions in `next-on-netlify` and make sure `next-aws-lambda` is included during bundling

**Comment by ehmicky** [comment]
Some PR at https://github.com/netlify/netlify-plugin-nextjs/pull/48 for the first point.

**Comment by lindsaylevine** [comment]
i just merged your PR and think we can close this issue. i just synced with finn and he's removing the next-aws-lambda dependency in a WIP PR: https://github.com/netlify/next-on-netlify/pull/85

proposal: 
1. close this issue
2. open another issue to remove the next-aws-lambda install once finn's PR is merged and published

let me know your thoughts! i defer to you @ehmicky

**Comment by ehmicky** [comment]
Yes, problem solved!
